lib/gpu/context.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of gpu;
```

Comment:
Since the flutter gpu stuff is being developed as its own package, you don't necessarily need to use part/part of to force it into a single library. You can follow a regular import/export pattern like the framework does and it will just work.

Reply:
I keep going back and forth on this TBH. I really like the idea of importing one unit as `gpu`. Not married to it, though.

Maybe we can get the best of both worlds by structuring everything into separately importable files, but have an extra one that exports all of the others. I could do something like that in a follow-up. Thoughts?

Reply:
You can absolutely do that with regular libraries, see https://github.com/flutter/flutter/blob/master/packages/flutter/lib/widgets.dart

You would create a single flutter_gpu/lib/flutter_gpu.dart library that re-exports the types you want to be public from flutter_gpu/lib/src/**

Reply:
In fact, should we be fully structuring this like a package in all respects, with a `lib` directory and a `package.yaml` that specifies the flutter SDK as a dependency?

Reply:
Generally yes - Do you depend on the flutter_sdk (package:flutter) or do you depend on dart:ui?

Reply:
My bad, I meant `sky_engine` in this case.

Reply:
Yeah, in that case I would add the following to the pubspec :

```
 sky_engine:
    sdk: flutter
 ```

Reply:
I reworked it to as an actual package and changed the import scheme.